### ISSUE TITLE
Share my desktop do not work

### ISSUE THREAD

**Issue by bubooo** [tieto/sipe#195]
I have moved to Mint 19 (18.4 ubuntu) with xfce and ppa:sipe-collab/ppa enabled. Audio calls work after playing with pavucontrol, but sharing desktop does not work. See attached debug. Just not sure if "bt full" should be issued before starting sharing. When mate from M$ Skype fB start sharing I approve it, just waiting for appearing remmina window, but nothing. If I initiate session he see nothing then final error message like "couldn't setup because media was not setup correctly". GDB show nothing at that time.
It is working on my Elementary Loki (16.04.2 Ubuntu) with  ppa:sipe-collab/stable with pidgin3.0 and pidgin-sipe 2.12 I guess. No additional setup was required.
[pidgin_share_desktop_debug.log](https://github.com/tieto/sipe/files/2588886/pidgin_share_desktop_debug.log)
[list_of_packages.txt](https://github.com/tieto/sipe/files/2588899/list_of_packages.txt)

**Comment by bubooo** [comment]
Elementary Loki pidgin-sipe version is 2.21.1

**Comment by xhaakon** [comment]
Logging was not enabled, so the debug file isn't useful. Please use this command line to run Pidgin:
```
GST_DEBUG=3 PURPLE_UNSAFE_DEBUG=1 G_MESSAGES_DEBUG=all NICE_DEBUG=all,libnice-nice-verbose pidgin --debug
```

Also, there's no need to run Pidgin inside gdb unless you're debugging a crash, which doesn't seem to be the case.

**Comment by bubooo** [comment]
ahhh, it is clear now when I read whole howto. Please provide e-mail 
address to be send log there as I don't want make some data public.

**Comment by xhaakon** [comment]
@bubooo Please use the email from my GitHub profile. (But you should already have it from previous conversation.)

**Comment by xhaakon** [comment]
In the log I've received, the problem looks like libnice not being able to reliably establish a connection with the other participant using a media relay server.

In order for us to have a clean starting point for debugging the issue, I've uploaded new `gstreamer1.0-nice` and `libnice10` packages that include the latest upstream changes and fixes into 18.04 PPA.

Please `apt-get upgrade` your installation and check if the updated library solves your problem. If you're still experiencing issues, send me the new log.

**Comment by bubooo** [comment]
I've sent you new log. First impression was great as looked like solved. 
But it was very hard to log in to Skype server as SSL handshare error.  
But all was working until crashed suddenly.

Therefore I've run debug and have enable more ciphers through NSS 
Preferencies plugin. Sign in was quicker, but voice quality decreased. 
Many debug messages like /
/

/0:19:09.971368239 15319 0x7fbadc115890 WARN pulse 
pulsesink.c:702:gst_pulsering_stream_underflow_cb:<pulsesink2> Got 
underflow//
//0:19:10.147556037 15319 0x7fbadc115890 WARN                   pulse 
pulsesink.c:702:gst_pulsering_stream_underflow_cb:<pulsesink2> Got 
underflow//
//0:19:11.541871314 15319 0x7fba9c64b9e0 WARN rtpjitterbuffer 
rtpjitterbuffer.c:570:calculate_skew: delta - skew: 0:00:04.087931123 
too big, reset skew/

Sharing desktop stopped working and pidgin become unstable

**Comment by xhaakon** [comment]
There's quite a lot going on in the new log file. I'm focusing on application sharing.

> But it was very hard to log in to Skype server as SSL handshare error. 
> But all was working until crashed suddenly.

libnice has nothing to do with the login process, so that SSL error can't be directly related to its upgrade.

I've however noticed another thing: during the first, working, appshare session, libnice reported three network interfaces - `lo`, `eno1` and `vpn0`, but after you logged in again and accepted the incoming presentation, there were only `lo` and `eno1`. Apparently your VPN had meanwhile crashed or you had turned it off. That could be the reason for those troubles with login, poor audio and failed remote desktop sharing if the VPN is needed for accessing your company network.

**Comment by bubooo** [comment]
yes, log is huge as it was not stable, I will try today produce shorter 
one with just single issue.

Regarding interfaces.  Initially I was connected through VPN but the 
voice quality was horrible and then session has crased. So that I have 
decided to turn off VPN, but voice quality didn't improved.

I will try create another log a last one can be interesting about voice 
quality degradation.

**Comment by bubooo** [comment]
I've sent newest debug logs when sharing desktop is still not working.

yes, there were many issues merged together. Initially I have been 
connected through SSL VPN into office so I have thought that bad voice 
quality is caused by additional encryption. So I have disconnected from 
VPN and start pidgin once more, but it didn't help.

Today's logs are without VPN. Voice quality is better, but was better 
with disabled NSS Preference plugin. Indeed there are changes on SKype 
server side as today I was able to connect in first round.
